networking_cisco/plugins/cisco/service_plugins/cisco_router_gbp_plugin.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2016 Cisco Systems, Inc.  All rights reserved.
```

Comment:
Note the class inheritance -- this inherits from l3_router_appliance_gbp_db.py, not l3_router_appliance_db.py . We needed a different implementation of the L3 APIs, which are kept in the DB/mixin class, so that's why there are two different DB classes -- which is why the plugin class has to reimplement everything. One alternative was to monkeypatch the existing DB class from a derived plugin, but that meant leaking DB pieces into the plugin class, which is why I chose duplicating the plugin code instead.